external-crates/move/crates/move-bytecode-verifier/src/type_safety.rs

```diff
@@ -140,14 +141,55 @@ pub(crate) fn verify<'a>(
 
     for block_id in function_view.cfg().blocks() {
         for offset in function_view.cfg().instr_indexes(block_id) {
-            let instr = &verifier.function_view.code().code[offset as usize];
-            verify_instr(verifier, instr, offset, meter)?
+            let code = &verifier.function_view.code();
+            let instr = &code.code[offset as usize];
+            let jump_tables = &code.jump_tables;
+            verify_instr(verifier, instr, jump_tables, offset, meter)?
         }
     }
 
     Ok(())
 }
 
+// Verifies:
+// * Top of stack is an immutable reference
+// * The type pointed to by the reference is the same as enum definition expected in as the "head
+//   constructor" for the jump table. This is important for exhaustivity.
+// * The variant tags in the jump table are both unique, and complete for the specified enum.
+fn variant_switch(
```

Comment:
Mmm probably.. probably. Will add.

Reply:
Looked at it and I'm not sure if we need/want to add metering here -- the metering for type_safety is all based on the number of types being pushed on the stack whereas this isn't doing any pushing. That + this function is fairly low overhead (no loops or anything like that). Thoughts?

Reply:
I think we should meter here to be safe. We essentially meter similarly when we push onto the stack for Unpack, but for variant switch, we have an O(NUM_VARIANT) operation with no metering.

In short, we don't meter in similar patterns because there are stack effects to catch the O(n) behavior. But here we have an O(n) without any metering

Reply:
Chatted out-of-band: this doesn't need to be metered since it's an O(1) op (since we don't need to iterate through the elements of the jump table/enum).